src/App/App.test.js

```diff
@@ -1,9 +1,152 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
 import App from './App';
+import '@testing-library/jest-dom';
+import { MemoryRouter, Router } from 'react-router-dom';
+import { sampleDay, sampleWeek, emptySampleWeek } from '../testData';
+import { getImageByDate, getAllImages } from '../APICalls.js';
+jest.mock('../APICalls');
+import MutationObserver from '@sheerun/mutationobserver-shim';
+window.MutationObserver = MutationObserver;
 
-test('renders learn react link', () => {
-  const { getByText } = render(<App />);
-  const linkElement = getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+describe('App render', () => {
+  it('should render a landing page on initialization', () => {
+    getAllImages.mockResolvedValue(sampleWeek)
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+
+    const landingImage = screen.getByAltText('Jupiter')
+
+    expect(landingImage).toBeInTheDocument()
+  })
+
+  it('should render the image of the data after 4 seconds', async() => {
+    getAllImages.mockResolvedValue(sampleWeek)
+    jest.useFakeTimers()
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    jest.runAllTimers()
+
+    const day1 = await waitFor(() => screen.getByAltText('Day1'))
+
+    expect(day1).toBeInTheDocument()
+  })
+});
+
+describe('App empty fetch render', () => {
+  it('should render a page with no image or information when the server returns no images', async() => {
+    getAllImages.mockResolvedValue(emptySampleWeek)
+    jest.useFakeTimers()
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    jest.runAllTimers()
+
+    const favoritesButton = await waitFor(() => screen.getByRole('link', {name: 'My Favorites'}))
+    const day1 = await waitFor(() => screen.queryByAltText('Day1'))
+
+    expect(favoritesButton).toBeInTheDocument()
+    expect(day1).not.toBeInTheDocument()
+  })
 });
+
+describe('App main page navigation', () => {
+  it('should render the next image when clicking the down & up arrows', async() => {
+    getAllImages.mockResolvedValue(sampleWeek)
+    jest.useFakeTimers()
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    jest.runAllTimers()
```

Comment:
jest.useFakeTimers() & jest.runAllTimers() is causing multiple renders with MemoryRouter